src/server/game/Grids/Notifiers/GridNotifiers.cpp

```diff
@@ -132,8 +132,21 @@ inline void CreatureUnitRelocationWorker(Creature* c, Unit* u)
         return;
 
     if (c->HasReactState(REACT_AGGRESSIVE) && !c->HasUnitState(UNIT_STATE_SIGHTLESS))
+    {
         if (c->IsAIEnabled && c->CanSeeOrDetect(u, false, true))
+        {
+            // If entering combat, but distracted, clear the state so engagement is immediate
+            if (u->GetTypeId() == TYPEID_PLAYER && c->HasUnitState(UNIT_STATE_DISTRACTED))
```

Comment:
While it works. I think it's quite dirty (and doesn't cover other methods of getting into combat).

I'm wondering if I should just clear MotionMaster in the 3 places AttackStart happens (the places I removed de-stealthing). I can't check for UNIT_STATE_DISTRACTED, I'm sure by that time it'd been cleared.